### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by RamVasuthevan**

### CODE REVIEW

File: bills.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                row[f"sponsor_title{i+1}"] = sponsor.title
+            writer.writerow(row)
+
+
+if __name__ == "__main__":
+    try:
+        html_content = extract(
+            ROOT_DOMAIN + "en/legislative-business/bills/parliament-42/session-2/"
+        )
+        parliament = "42nd Parliament, Session 2"
```
Comment (kyleyarwood) [inline bills.py]: Assuming this was just for testing since it seems like you use variables in `main.py`

File: bills.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                "div", class_="field--name-field-sponsor-role"
+            )
+            sponsor_title = (
+                sponsor_title_element.text.strip() if sponsor_title_element else ""
+            )
+            bill_sponsors.append(SponsorInfo(name=sponsor_name, title=sponsor_title))
+
+        bill_info_list.append(
+            BillInfo(parliament, bill_number, bill_title, url_title, bill_sponsors)
+        )
```
Comment (kyleyarwood) [inline bills.py]: Really like the use of dataclasses here

File: bills.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+        writer.writeheader()
+
+        for bill in bills_data:
+            row = {
+                "parliament": bill.parliament,
+                "bill_number": bill.bill_number,
+                "bill_title": bill.bill_title,
+                "url_title": bill.url_title,
```
Comment (kyleyarwood) [inline bills.py]: Is there a nicer way of creating a dict for {attribute_name: attribute_value}? it looks fine, I'm just curious if there was an easier way for something that has a lot more fields. It'd also probably be hard to do that with the sponsors list here

File: bills.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+def extract(parliament_url) -> str:
+    response = requests.get(parliament_url)
+    response.raise_for_status()
+    return response.text
+
+
+def transform(parliament: str, html_content: str) -> List[BillInfo]:
+    soup = BeautifulSoup(html_content, "html.parser")
+    table = soup.find("table", class_="views-table views-view-table cols-3")
```
Comment (kyleyarwood) [inline bills.py]: God bless BeautifulSoup

File: parliaments.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+
+if __name__ == "__main__":
+    try:
+        html_content = extract()
+        extracted_data = transform(html_content)
+        load(extracted_data)
+        print(f"Data extracted, processed, and saved to '{CSV_FILENAME}' successfully!")
+    except requests.RequestException as e:
+        print(f"An error occurred during extraction: {e}")
```
Comment (kyleyarwood) [inline parliaments.py]: Again assuming this is just for testing?

File: main.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import parliaments
+import bills
+
+
+def main():
+    parliament_html_content = parliaments.extract()
```
Comment (kyleyarwood) [inline main.py]: do you want to try and except for requests exceptions here? and on line 15?

**Review by kyleyarwood — COMMENTED** [review]
Overall looks great, a couple small comments. Have you thought about using a [logging.Logger](https://docs.python.org/3/library/logging.html#logger-objects) for logging to make it easier to debug if anything fails?

File: bills.py (hunk 1 above)
Comment (RamVasuthevan) [reply bills.py]: Yeah, this is if I just want to run the file

File: main.py (hunk 6 above)
Comment (RamVasuthevan) [reply main.py]: Hmm, I think I'll remove the try/except from the other two files. This code will only run once a month and should not have any non-network-related exceptions for years.

File: bills.py (hunk 3 above)
Comment (RamVasuthevan) [reply bills.py]: Yeah, I don't think there's a cleaner way to do it. Usually, I'd be down for some Lambda madness, but I think this much more readable

**Pull request merged**